### ISSUE TITLE
Track stats on installed patterns

### ISSUE THREAD

**Issue by iandunn** [WordPress/pattern-directory#24]
Similar to the plugin directory, it'd be useful for the pattern directory to track stats on which patterns are installed. 

That could help the w.org front-end directly display popular patterns, etc.

This could be done by sending a list of installed pattern IDs to api.w.org when Core checks for updates, which is how it's done for plugins.

It's not clear yet how patterns will be "installed", since inserting a pattern into a post could detach it from the canonical version on w.org. The designs might include a way of "favoriting" a w.org pattern, or similar.

**Comment by ryelle** [comment]
We can't really track installs, since patterns are more ephemeral - you don't really install them. We could track:

- Number of favorites
- Number of times (and reasons) reported
- Number of times copied(? - I know how to do this in calypso, but not on wporg)

I'm taking this out of the "initial launch" milestone - while important, it wouldn't add or detract from the stability of the directory.

**Comment by iandunn** [comment]
Those sound good 👍🏻 

> Number of favorites
> Number of times (and reasons) reported

I'm guessing we've already got that data, by nature of it.  Eventually we might want to build a public report like we have for WordCamp data, but the first step would just be collecting the important stuff early.

> Number of times copied

This seems like a pretty important one, to show usage. I'd guess most of the installs would be done in Core, which we don't track, but the installs on w.org may be highly correlated?

if we save installs per-pattern, we could compare usage across locales.

It might be nice to have a snapshot of # of installs per month, so we can see usage trends over time?

**Comment by iandunn** [comment]
Once Gutenberg starts allowing searches of remote patterns, then tracking the most popular terms seems helpful. we could use that to identify gaps where patterns that people want don't exist yet.

cc @shaunandrews ,  @andreamiddleton, @mtias in case y'all have any other ideas for stats that'd be useful to track

**Comment by mtias** [comment]
Favorites is likely going to be the best metric to rely on. Once we have a flow that doesn't involve copy / paste that would also be a good way to get a sense of installs.

**Comment by iandunn** [comment]
> Once we have a flow that doesn't involve copy / paste

Is that because copying doesn't guarantee that they'll paste & publish? I'd guess that it'd still be highly correlated.

**Comment by StevenDufresne** [comment]
> Is that because copying doesn't guarantee that they'll paste & publish? I'd guess that it'd still be highly correlated.

I think the challenge will be that people may browse on `.org` but install through Gutenberg.

.... Maybe we should add a copy button next to the title, like how GitHub does for branch names 🤔 .

**Comment by mtias** [comment]
> Is that because copying doesn't guarantee that they'll paste & publish? I'd guess that it'd still be highly correlated.

Yes, you might click copy even if you don't have a WordPress install just out of curiosity. It feels like a very indirect measure.

**Comment by ryelle** [comment]
Does that mean that the method of using a pattern will change in core? Whether it's copied from wp.org or added from the inserter in the editor, it's only content added to the post.

---

What I'm hearing right now, for stats tracking on launch, is snapshot-style record keeping (using [the 5ftf cron setup?](https://github.com/WordPress/five-for-the-future/blob/production/plugins/wporg-5ftf/includes/stats.php)) to track total number of favorite (and reported?) patterns.

Post launch, when the core feature is iterated on, we will want to track search terms, and however patterns are officially installed.

**Comment by mtias** [comment]
> Whether it's copied from wp.org or added from the inserter in the editor, it's only content added to the post.

It won't change, but an action that performs insertion is an action that ensure selection + insertion happens, closer to an "activation".

> What I'm hearing right now, for stats tracking on launch, is snapshot-style record keeping

That'd be my inclination as a start, it should be something that can be done on the patterns repository itself without having to deal with user actions.
